source/frontend/src/features/leases/detail/LeasePages/improvements/AddImprovementsForm.tsx

```diff
@@ -51,9 +52,29 @@ export const AddImprovementsForm: React.FunctionComponent<
                         const title =
                           sectionTitles.get(improvement.propertyImprovementTypeId) ?? 'N/A';
                         const nameSpace = `improvements.${index}`;
+
+                        const HeaderDisplayNode = (typeId: string): ReactNode => {
```

Comment:
In this case I think this duplication is fine, but let's try and use reusable components where appropriate to reduce duplication and reduce the risk of view/edit visual differences.